lib/Controller/RecoveryController.php

```diff
@@ -89,12 +89,12 @@ public function adminRecovery($recoveryPassword, $confirmPassword, $adminEnableR
 				Http::STATUS_BAD_REQUEST);
 		}
 
-		if (isset($adminEnableRecovery) && $adminEnableRecovery === '1') {
+		if ($adminEnableRecovery !== null && $adminEnableRecovery === '1') {
 			if ($this->recovery->enableAdminRecovery($recoveryPassword)) {
 				return new DataResponse(['data' => ['message' => (string)$this->l->t('Recovery key successfully enabled')]]);
 			}
 			return new DataResponse(['data' => ['message' => (string)$this->l->t('Could not enable recovery key. Please check your recovery key password!')]], Http::STATUS_BAD_REQUEST);
-		} elseif (isset($adminEnableRecovery) && $adminEnableRecovery === '0') {
+		} elseif ($adminEnableRecovery !== null && $adminEnableRecovery === '0') {
```

Comment:
```suggestion
		} elseif ($adminEnableRecovery === '0') {
```
The way it is now looks a bit silly - if `$adminEnableRecovery === '1'` then it is sure to be not null.